packages/@ember/array/index.ts

```diff
@@ -250,86 +250,13 @@ function mapBy<T>(this: EmberArray<T>, key: string) {
   @public
 */
 interface EmberArray<T> extends Enumerable {
-  length: number;
-  objectAt(idx: number): T | undefined;
-  objectsAt(indexes: number[]): Array<T | undefined>;
-  firstObject: T | undefined;
-  lastObject: T | undefined;
-  slice(beginIndex?: number, endIndex?: number): NativeArray<T>;
-  indexOf(object: T, startAt?: number): number;
-  lastIndexOf(object: T, startAt?: number): number;
-  forEach<Target>(
-    callback: (this: Target, item: T, index: number, arr: this) => void,
-    target?: Target
-  ): this;
-  getEach<K extends string>(key: K): NativeArray<Value<T, K>>;
-  setEach<K extends string>(key: K, value: Value<T, K>): this;
-  map<U, Target>(
-    callback: (this: Target, item: T, index: number, arr: this) => U,
-    target?: Target
-  ): NativeArray<U>;
-  mapBy<K extends string>(key: K): NativeArray<Value<T, K>>;
-  filter<Target>(
-    callback: (this: Target, item: T, index: number, arr: this) => unknown,
-    target?: Target
-  ): NativeArray<T>;
-  reject<Target>(
-    callback: (this: Target, item: T, index: number, arr: this) => unknown,
-    target?: Target
-  ): NativeArray<T>;
-  filterBy(key: string, value?: unknown): NativeArray<T>;
-  rejectBy(key: string, value?: unknown): NativeArray<T>;
-  find<S extends T, Target = void>(
-    predicate: (this: void, value: T, index: number, obj: T[]) => value is S,
-    thisArg?: Target
-  ): S | undefined;
-  find<Target = void>(
-    callback: (this: Target, item: T, index: number, arr: this) => unknown,
-    target?: Target
-  ): T | undefined;
-  findBy<K extends string>(key: K, value?: Value<T, K>): T | undefined;
-  every<Target = void>(
-    callback: (this: Target, item: T, index: number, arr: this) => unknown,
-    target?: Target
-  ): boolean;
-  isEvery<K extends string>(key: K, value?: Value<T, K>): boolean;
-  any<Target = void>(
-    callback: (this: Target, item: T, index: number, arr: this) => unknown,
-    target?: Target
-  ): boolean;
-  isAny<K extends string>(key: K, value?: Value<T, K>): boolean;
-  reduce<V>(
-    callback: (summation: V, current: T, index: number, arr: this) => V,
-    initialValue?: V
-  ): V;
-  invoke<K extends string>(
-    methodName: K,
-    ...args: Value<T, K> extends AnyFn ? Parameters<Value<T, K>> : unknown[]
-  ): NativeArray<Value<T, K> extends AnyFn ? ReturnType<Value<T, K>> : unknown>;
-  toArray(): T[];
-  compact(): NativeArray<Exclude<T, null>>;
-  includes(object: T, startAt?: number): boolean;
-  sortBy(key: string): T[];
-  uniq(): NativeArray<T>;
-  uniqBy(key: string): NativeArray<T>;
-  without(value: T): NativeArray<T>;
-}
-const EmberArray = Mixin.create(Enumerable, {
-  init() {
-    this._super(...arguments);
-    setEmberArray(this);
-  },
-
   /**
     __Required.__ You must implement this method to apply this mixin.
 
     Your array must support the `length` property. Your replace methods should
     set this property whenever it changes.
-
-    @property {Number} length
-    @public
```

Comment:
We need to preserve these annotations in the switchover! They're used by the doc generation tool in knowing (a) what to emit (`@public`) and (b) how to render it in the docs (the type, in particular).

Reply:
Thanks for clarifying, Chris. This PR initially had those annotations earlier but I removed it later after referring to similar PRs. I'll add them back now.

Reply:
Yeah, not your fault at all—thank you!